spot-client/src/spot-tv/ui/components/meeting-frame/ApiHealthCheck.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Checks if the Jitsi API is responsive.
+ */
+export default class ApiHealthCheck {
+    /**
+     * The amount of time to wait in milliseconds for the Jitsi API to return
+     * a response.
+     */
+    static pingTimeLimitMs = 5000;
+
+    /**
+     * The amount of time to wait in milliseconds before sending a new request
+     * into the Jitsi API.
+     */
+    static pingIntervalMs = 5000;
+
+    /**
+     * Initializes a new {@code ApiHealthCheck} instance.
+     *
+     * @param {Object} jitsiApi - An instance of JitsiMeetExternalAPI.
+     * @param {Function} onError - Callback to invoke when a health check fails.
+     */
+    constructor(jitsiApi, onError) {
+        this._jitsiApi = jitsiApi;
+        this._onError = onError;
+
+        this._isRunning = false;
+    }
+
+    /**
+     * Begins polling the Jitsi API to check if the iFrame is responsive.
+     *
+     * @returns {void}
+     */
+    start() {
+        if (this._isRunning) {
+            throw new Error('Destroyed. Create new instance.');
```

Comment:
that is not quite accurate, it is not destroyed, but already running which could be a no-op maybe?

Reply:
Right. Bug from the `rename destroy to stop` change.